### DISCUSSION TITLE
Koishi 4.5.0

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#587] [General]
## Notable Changes

- **orm:** 新增了 @koishijs/orm，其中包含了 @koishijs/core 中所用到的 ORM 逻辑 (95a1155548f95764749588c7262f622e216a3940, 7267d09ea1464b5e7a2ef3c34770f8ee0c3912ba)
- **orm:** 支持了通过路径折叠的方式定义嵌套的属性，有利于开发者解耦和封装插件逻辑 (cf23ba2b19c0a89a768e47d7c60da28e88cb5019, 221dd87379cb430c1f1ebc4ed9fa56cff586e1bb, e4e17bc4dcc80f202433dbe9f18b7639c51f09ab, 0801a1541a7979e8d139b3e71a32c9940e1a9e5a, 87da7097774700a70618ff3f0130e418792f96e3, dbef5ccb5eeb4945704345118399225e210dcb1b, f6848efe98d14075c8a49f4f49847454a93b59be)
- **orm:** 支持了定义和扩展 model 原型链上的方法，为用户封装数据库的操作提供了方便 (af906ca0254329ad9584b084cb9ce9ebdd342992, ce7707778356455e07b0e5f9114bdb5b650ffd9a)

## Features

- **core:** i18n 服务现在也支持热重载了 (3fa555eaef91896666bef9d9dee0107adeb71a7b)
- **core:** 支持了多语言的指令使用方法和使用示例  (6e3c4a69617c6a8cf4e7ae2b4142d2a389fbda22)
- **mysql:** 优化了 `update` 语句的生成行为，现在能生成更简洁的 SQL 了 (4893f2e441eb320f412f4c3b17e8fd0b8ac52e21)
- **telegram:** 支持了 [Telegram Bot API](https://github.com/tdlib/telegram-bot-api) 的文件下载 (#551) (89354a260e7a016c66d30edf5396ef9fb2842e4f)

## Bug Fixes

- **auth:** 调整了令牌更新逻辑，支持了多端登录 (17b73186416de6ac58c724fc093793edc1ae8094)
- **console:** 修复了 console 服务被卸载后可能的报错问题 (25b60ffb5e80f259bdcbd41ac3a66e1e4f12e227)
- **core:** 修复了立即注册服务在插件初始化前就生效的问题 (#571) (af817b2e7c3907e5193dfe8a2a98e0ae25c961ec)
- **discord:** 修复了始终处于重连状态的问题 (0c8fd6eb5bce67b56c53a32c4198a292e2e6e77c)
- **telegram:** 修复了无法获取引用消息的问题 (037ad24e5e5a799140c3c643906d6db4bad636f5)

## Other Changes

- **switch:** basic implementation (9cdce19c0537ecad4694ac043ab427df6f76e0cc)
- docs: add qq group link (77673accf0ff65b39bc00cb06073efacadbbe238)
- build: enhance build scripts (e4d9b53b7ed610785badc227d748084ed123578f)
- chore: bump versions (15ffd665cb3363f28761599fdd17fe11607e3390)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.5.0'>Koishi 4.5.0</a>.</em>

**Comment by DGCK81LNN** [comment]
?不懂就问，怎么npm上没看到更新

**Reply by shigma** [reply]
更了更了（昨天脑抽

**Reply by DGCK81LNN** [reply]
好耶，这就update）
